micro-rdk/src/common/data_store.rs

```diff
@@ -54,7 +55,18 @@ lazy_static::lazy_static! {
     static ref DATA_STORE_INITIALIZED: AtomicBool = AtomicBool::new(false);
 }
 
+/// A trait for sn entity that is capable of reading from a store without consuming
+/// the messages until a command to flush the read messages is sent
+pub trait DataStoreReader {
```

Comment:
If I was a whole lot more proficient with Rust I think I'd be trying to figure out how to make this work with an Iterator that yielded back each `BytesMut`, but as I don't know how to produce such a thing, I don't think it is fair to ask you to do it either ;)

Reply:
I'll give it a shot

Reply:
I have an idea that works, but I want to discuss it with you tomorrow before I move forward